.github/workflows/codequality.yml

```diff
@@ -0,0 +1,38 @@
+name: Code Quality
```

Comment:
I put those steps into a reusable workflow because I am expecting to reuse it in the master pipeline, before the deployment

Reply:
Interesting approach; not sure what to think of it though. The re-usable workflow runs a lot of duplicate commands

Reply:
I also tend to have two pipelines, but I run the codequality, unit and e2e stages again in master. It sounds a bit redundant but it gives me more security. For example, you won't test anything of the checks we do when you push straight to main branch. I use CI to give me that extra layer of security. I also like the workflow diagrams because they are very easy to undestand.  

![Screenshot 2022-09-07 at 10 46 19](https://user-images.githubusercontent.com/32243459/188847964-5f3283b9-d939-466a-ae45-2994c2aa31b1.png)

In summary, the pipeline can be a bit slower to deploy in prod, but for a small project like this I don't expect to make a noticable difference.

Reply:
We can try it out, see how it works.